pkg/tsdb/prometheus/resource/resource.go

```diff
@@ -55,113 +49,49 @@ func delStopHeaders(header http.Header) {
 	}
 }
 
-func addHeaders(header http.Header, toAdd map[string]string) {
-	for k, v := range toAdd {
-		header.Add(k, v)
-	}
-}
-
-func normalizeReqHeaders(headers map[string][]string) map[string]string {
-	h := make(map[string]string, len(headers))
-	for k, v := range headers {
-		h[k] = strings.Join(v, ",")
-	}
-	return h
-}
-
 func New(
-	httpClientProvider httpclient.Provider,
-	cfg *setting.Cfg,
-	features featuremgmt.FeatureToggles,
+	httpClient *http.Client,
 	settings backend.DataSourceInstanceSettings,
 	plog log.Logger,
 ) (*Resource, error) {
-	var jsonData map[string]interface{}
-	if err := json.Unmarshal(settings.JSONData, &jsonData); err != nil {
-		return nil, fmt.Errorf("error reading settings: %w", err)
-	}
-
-	customHeaders := make(map[string]string)
-	var jsonDataMap map[string]interface{}
-
-	err := json.Unmarshal(settings.JSONData, &jsonDataMap)
+	jsonData, err := utils.GetJsonData(settings)
 	if err != nil {
 		return nil, err
 	}
-
-	index := 1
-	for {
-		headerNameSuffix := fmt.Sprintf("httpHeaderName%d", index)
```

Comment:
Custom headers are already handled in http client creation, but before there was a bug where we overwritten the headers which were supplied.